src/r_draw.c

```diff
@@ -868,7 +868,7 @@ void R_DrawBorder (int x, int y, int w, int h, int s)
   for (i = 0; i < w; i += 8)
     V_DrawPatch(x + i - WIDESCREENDELTA, y - 8, s, patch);
 
-  patch = W_CacheLumpName("brdr_b", PU_CACHE);
+  patch = W_CacheLumpName("brdr_b", PU_STATIC);
```

Comment:
Could you add a comment why this one needs to be special, please?

Reply:
Fixed by https://github.com/fabiangreffrath/woof/pull/1225/commits/070f291d451c197b77886b8fdd7ab544871a68ad

Reply:
This `PU_STATIC`/`PU_CAHE` dynamic with a global variable looks confusing to me. I suggest caching bezel height in `st_stuff.c` instead of the patch:

<details>

```diff
diff --git a/src/r_draw.c b/src/r_draw.c
index 10a3cd4..d7459c7 100644
--- a/src/r_draw.c
+++ b/src/r_draw.c
@@ -868,9 +868,7 @@ void R_DrawBorder (int x, int y, int w, int h, int s)
   for (i = 0; i < w; i += 8)
     V_DrawPatch(x + i - WIDESCREENDELTA, y - 8, s, patch);
 
-  // PU_CACHE is freed on every level load. Use PU_STATIC to retain the pointer
-  // address for both the reduced screen size and widescreen status bar borders.
-  patch = W_CacheLumpName("brdr_b", PU_STATIC);
+  patch = W_CacheLumpName("brdr_b", PU_CACHE);
   for (i = 0; i < w; i += 8)
     V_DrawPatch(x + i - WIDESCREENDELTA, y + h, s, patch);
 
diff --git a/src/st_stuff.c b/src/st_stuff.c
index f5099b4..f885c52 100644
--- a/src/st_stuff.c
+++ b/src/st_stuff.c
@@ -236,7 +236,7 @@ static patch_t *faceback[MAXPLAYERS]; // killough 3/7/98: make array
 static patch_t *armsbg;
 
 // Bezel bottom edge for st_solidbackground.
-static patch_t *bezel;
+static int bezel_height;
 
 // weapon ownership patches
 static patch_t *arms[6][2];
@@ -416,10 +416,12 @@ void ST_refreshBackground(boolean force)
           // [crispy] preserve bezel bottom edge
           if (scaledviewwidth == SCREENWIDTH)
           {
+            patch_t *const patch = W_CacheLumpName("brdr_b", PU_CACHE);
+
             for (x = 0; x < WIDESCREENDELTA; x += 8)
             {
-              V_DrawPatch(x - WIDESCREENDELTA, 0, BG, bezel);
-              V_DrawPatch(ORIGWIDTH + WIDESCREENDELTA - x - 8, 0, BG, bezel);
+              V_DrawPatch(x - WIDESCREENDELTA, 0, BG, patch);
+              V_DrawPatch(ORIGWIDTH + WIDESCREENDELTA - x - 8, 0, BG, patch);
             }
           }
         }
@@ -1084,7 +1086,10 @@ void ST_loadGraphics(void)
   have_xdthfaces = i;
 
   // Bezel bottom edge for st_solidbackground.
-  bezel = W_CacheLumpName("BRDR_B", PU_STATIC);
+  {
+    patch_t *const patch = W_CacheLumpName("BRDR_B", PU_CACHE);
+    bezel_height = patch->height;
+  }
 }
 
 void ST_loadData(void)
@@ -1313,8 +1318,8 @@ static int StatusBarBufferHeight(void)
   if (armsbg && armsbg->height > st_height)
     st_height = armsbg->height;
 
-  if (bezel && bezel->height > st_height)
-    st_height = bezel->height;
+  if (bezel_height > st_height)
+    st_height = bezel_height;
 
   for (i = 0; i < MAXPLAYERS; i++)
   {
@@ -1359,10 +1364,10 @@ void ST_Warnings(void)
                          "Expected <= %d.", armsbg->height, ST_HEIGHT);
   }
 
-  if (bezel && bezel->height > ST_HEIGHT)
+  if (bezel_height > ST_HEIGHT)
   {
     I_Printf(VB_WARNING, "ST_Init: Non-standard BRDR_B height of %d. "
-                         "Expected <= %d.", bezel->height, ST_HEIGHT);
+                         "Expected <= %d.", bezel_height, ST_HEIGHT);
   }
 
   for (i = 0; i < MAXPLAYERS; i++)
```
</details>

Reply:
Okay new changes make everything `PU_CACHE` and is more defensive.